client/src/SessionController.js

```diff
@@ -11,25 +19,56 @@ SessionController.propTypes = {
 
 export function SessionController({ children }) {
   const dispatch = useDispatch()
-  const tripsStatus = useSelector((state) => state.trip.status)
+  const navigate = useNavigate()
+  const tripsStatus = useSelector((state) => state.trips.status)
+  const user = useSelector((state) => state.users.user)
+  const usersStatus = useSelector((state) => state.users.status)
   const [isLoading, setIsLoading] = useState(false)
   const [isMinimumLoadingTimeMet, setIsMinimumLoadingTimeMet] = useState(false)
 
   useEffect(() => {
-    if (tripsStatus === REQUEST_STATE.IDLE) {
-      dispatch(fetchTripsAsync())
+    if (usersStatus === REQUEST_STATE.LOGGINGOUT) {
+      dispatch(resetTrips())
+      dispatch(resetPreferences())
+      dispatch(resetView())
+      dispatch(resetMap())
+      dispatch(resetModalPreferences())
+      dispatch(updateStatus())
     }
-  }, [dispatch, tripsStatus])
+  }, [dispatch, usersStatus])
 
   useEffect(() => {
-    if (tripsStatus === REQUEST_STATE.WRITING) {
+    if (usersStatus === REQUEST_STATE.LOGGEDIN) {
+      setTimeout(() => {
+        navigate('/home')
+        dispatch(setAppView(AppView.NEW_TRIP))
+      }, 3000)
+    }
+  }, [dispatch, navigate, usersStatus])
+
+  useEffect(() => {
+    if (
+      tripsStatus === REQUEST_STATE.IDLE &&
+      usersStatus === REQUEST_STATE.LOGGEDIN
+    ) {
+      dispatch(fetchTripsAsync(user.id))
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [dispatch, tripsStatus, usersStatus])
+
+  useEffect(() => {
```

Comment:
if you are talking about line 59-71, its because this one keeps track of loading after user has logged in (updating or deleting trips) or when a user tries to log in (such operations have 3s+ delays in between). the one in line 43 is (going to be) just to redirect user to /home exactly when the user has successfully logged in.

Reply:
if you are talking about line 49-57. its because i want to make sure trips is idle (completely new). i think combining it into 43 would be fine, but not sure

Reply:
ah i meant the first one, was just wondering why both had a 3000 timeout and some kind of "logged in" check. thats all good now